File: components/string-utilities/src/main/java/org/cloudfoundry/credhub/utils/EmptyStringToNull.java
package org.cloudfoundry.credhub.utils;

import java.io.IOException;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;
import com.fasterxml.jackson.databind.JsonNode;

public class EmptyStringToNull extends JsonDeserializer<String> {

  @Override
  public String deserialize(final JsonParser jsonParser, final DeserializationContext context)
    throws IOException {
    final JsonNode node = jsonParser.readValueAsTree();
    if (node.asText().isEmpty()) {
      return null;
    }
    return node.textValue();
  }

}

File: components/string-utilities/src/main/kotlin/org/cloudfoundry/credhub/utils/EmptyStringToNull.kt
package org.cloudfoundry.credhub.utils

import tools.jackson.core.JsonParser
import tools.jackson.databind.DeserializationContext
import tools.jackson.databind.JsonNode
import tools.jackson.databind.ValueDeserializer
import java.io.IOException

class EmptyStringToNull : ValueDeserializer<String>() {
    @Throws(IOException::class)
    override fun deserialize(
        jsonParser: JsonParser,
        context: DeserializationContext,
    ): String? {
        val node = jsonParser.readValueAsTree<JsonNode>()
        if (node.asString().isEmpty()) {
            return null
        } else {
            return node.stringValue()
        }
    }
}
